src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsEventListener.java

```diff
@@ -31,13 +41,32 @@ public void onEvent(Event event) {
             default:
                 PrometheusExporter.instance().recordGenericEvent(event);
         }
+
+        setSessions(session.realms().getRealm(event.getRealmId()));
```

Comment:
does this have to be executed for every event? Maybe there are certain events that indicate a change in sessions like logins / logouts?

Reply:
Our Keycloak cluster is not configured to replicate sessions on all nodes.

If the Login is on one node, and the logout on an other. Until a new login or logout on the fist node, the sessions metrics is not updated for this node.

It's the reason why there is no filter on event's type.
I don't found a better solution to update the metrics in our cluster.